### PR TITLE
Ruler: Disable x-scope-orgid header append in remote write

### PR DESCRIPTION
**Pull request by jburnham**
**What this PR does / why we need it**:
 Adds option to disable addition of X-Scope-OrgID header in ruler remote write config. Needed to remote write to Amazon Managed Prometheus (Cortex).

**Which issue(s) this PR fixes**:
I suspect (but cannot prove) this will add a workaround for this existing open issue [#9438](https://github.com/grafana/loki/issues/9438)

**Special notes for your reviewer**:
This is a similar [PR from the Tempo project](https://github.com/grafana/tempo/pull/2974) where it could not remote write to Amazon Managed Prometheus. We were having the same issues with the X-Scope-OrgID header that now works once this header is disabled.
[Related issue from Tempo project](https://github.com/grafana/tempo/issues/2973)

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [X] Documentation added
- [X] Tests updated
- [x] `CHANGELOG.md` updated
  - [ ] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)
- [ ] If the change is deprecating or removing a configuration option, update the `deprecated-config.yaml` and `deleted-config.yaml` files respectively in the `tools/deprecated-config-checker` directory. [Example PR](https://github.com/grafana/loki/pull/10840/commits/0d4416a4b03739583349934b96f272fb4f685d15)

### CODE REVIEW

**Comment by jburnham** [comment]
Had an open ticket with AWS about this behavior and received the following reply.

> The AMP team has confirmed this is expected behavior, and applications should remove this header from their remote write requests to AMP.

They then also linked to this PR.

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
 
 ##### Enhancements
 
+* [11819](https://github.com/grafana/loki/pull/11819) **jburnham**: Ruler: Disable x-scope-orgid header append in remote write
```
Comment (dannykopping) [inline CHANGELOG.md:9]:
```suggestion
* [11819](https://github.com/grafana/loki/pull/11819) **jburnham**: Ruler: Add the ability to disable the `X-Scope-OrgId` tenant identification header in remote write requests.
```

**Review by dannykopping — APPROVED** [review]
LGTM, with a minor nit.

**Comment by jburnham** [comment]
@dannykopping I ran `make format` to fix the Drone CI check.

**Comment by dannykopping** [comment]
Thanks for your contribution @jburnham 👍

**Pull request merged**